addons/point_of_sale/models/pos_order.py

```diff
@@ -1042,6 +1087,152 @@ def refund(self):
             'target': 'current',
         }
 
+    @api.model
+    def remove_from_ui(self, server_ids):
+        """ Remove orders from the frontend PoS application
+
+        Remove orders from the server by id.
+        :param server_ids: list of the id's of orders to remove from the server.
+        :type server_ids: list.
+        :returns: list -- list of db-ids for the removed orders.
+        """
+        orders = self.search([('id', 'in', server_ids)])
+        for o in orders:
+            o.write({'state': 'cancel'})
+            o.sudo().with_context(force_company=self.env.user.company_id.id).unlink()
+        return orders.read(['id'])
+
+    def _create_or_update_order(self, order, draft = False, existing_order = False):
+        """ Create an order from an given dictionary.
+
+        This function creates or update an order from an given dictionary.
+
+        :param order: representation of an order.
+        :type order: dict.
+        :param draft: Indicate that the pos_order is not validated yet.
+        :type draft: bool.
+        :param existing_order: order to be updated or False.
+        :type existing_order: pos.order.
+        :returns: int -- id of created/updated pos_order.
+        """
+        to_invoice = order['to_invoice'] if not draft else False
+        order = order['data']
+        if to_invoice:
+            self._match_payment_to_invoice(order)
+        pos_order = self._process_order(order, draft, existing_order)
+
+        if not draft:
+            self._set_pos_order_paid(pos_order)
```

Comment:
maybe to check if an order is paid, we should base it on payment lines amount, and not on a draft argument passed in parameter.

Reply:
This would make the backend unaware if the order is finalised in the session, resulting in tickets that are not printed. In some countries (for instance Italy) you can get fined if you don't have a ticket when leaving a store / restaurant.